src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -207,11 +435,107 @@ impl KVEngine for EtcdKVEngine {
+
+        let self_prefix = prefix.to_string_key();
+
+        // Spawn a new task to handle the watch stream
+        TASK_MANAGER.spawn(TaskName::EtcdKeepAlive, |token| async move {
+            loop {
+                tokio::select! {
+                    () = token.cancelled() => {
+                        // Cancel this watch stream
+                        match watcher.cancel().await.with_context(|| "failed to cancel etcd watcher") {
+                            Ok(o) => {
+                                info!("etcd watcher task canceled, {o:?}");
+                            }
+                            Err(e) => {
+                                error!("failed to cancel etcd watcher, error={e:?}");
+                            }
+                        }
+                        break;
+                    }
+                    response = watch_stream.message() => {
+                        if let Ok(Some(watch_response)) = response {
+                            for event in watch_response.events() {
+                                // Get event data
+                                if let Some(kv) = event.kv() {
+                                    // Get key and value
+                                    let item_key = match kv.key_str() {
+                                        Ok(key) => if let Some(key) = key.strip_prefix(self_prefix.as_str()) { key.to_owned() } else {
+                                            error!("failed to strip prefix from key");
+                                            continue;
+                                        }
+                                        Err(e) => {
+                                            error!("failed to get key from etcd watch event, error={e:?}");
+                                            continue;
+                                        }
+                                    };
+                                    let item_value = match serde_json::from_slice(kv.value()) {
+                                        Ok(value) => value,
+                                        Err(e) => {
+                                            error!("failed to deserialize value from etcd watch event, error={e:?}");
+                                            continue;
+                                        }
+                                    };
+                                    match event.event_type() {
+                                        etcd_client::EventType::Put => {
+                                            match tx.send((item_key, Some(item_value))).await {
+                                                Ok(o) => {
+                                                    debug!("etcd watcher put event, res={o:?}");
+                                                }
+                                                Err(e) => {
+                                                    error!("failed to send etcd watcher put event, error={e:?}");
```

Comment:
Ignore tx.send() error case, no need for error log.  since rx is hold by caller code

Reply:
Ok, I have fixed it.